source/adapters/cuda/image.cpp

```diff
@@ -675,14 +675,15 @@ UR_APIEXPORT ur_result_t UR_APICALL urBindlessImagesImageCopyExp(
           // This should be unreachable.
           return UR_RESULT_ERROR_INVALID_VALUE;
         }
+        cuStreamSynchronize(Stream);
```

Comment:
Can we move this out of the conditional statements and just have it once after?

Reply:
done